ui/components/multichain/import-tokens-modal/import-tokens-modal.js

```diff
@@ -348,6 +362,38 @@ export const ImportTokensModal = ({ onClose }) => {
     dispatch(
       setPendingTokens({ customToken, selectedTokens, tokenAddressList }),
     );
+
+    const tempTokensToAdd = {
+      ...selectedTokens,
+      ...(customToken?.address && {
+        [customToken.address]: {
+          ...customToken,
+        },
+      }),
+    };
+
+    const tmpTokens = Object.values(tempTokensToAdd);
+    const tmpTokensToDispatch = tmpTokens.filter(
+      (elm) =>
+        contractExchangeRates?.[toChecksumHexAddress(elm.address)] ===
+        undefined,
+    );
+
+    const tokenAddresses = tmpTokensToDispatch.map((obj) => obj.address);
+    if (tmpTokensToDispatch.length !== 0) {
+      try {
+        const result = await fetchTokenContractExchangeRates({
```

Comment:
I don't think we should call controller method directly in UI layer, it should be in a selector or state management in background layer. Maybe it is better we move this to a hook or util function to make sure DOM is clean. we also should use `useCallBack` for such api calls to avoid extra render

Reply:
After our discussion; moved it to utils :D 🙏